### PR TITLE
Laggy DragNDrop fix

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR addes drag effects for every atom in the game. Usually if you drag something, it would activate it's Click method.

## Why It's Good For The Game

Have you ever seen your interface not reacting to your clicks? You press RMB to shoot that vampire, but instead nothing happends? You press three times on your stechkin in suit slot to pull it out? Your ability just ignores your mouse pressing? 
I thought i am the only one with that problem before due to my ping/packetloss/whatever, but im not

So, whats happening then? For some reason sometimes game registeres your clicks as DragNDrop
You can see it there https://youtu.be/jUhmdNjufwA
There is few items/objects that can be interacted with drag and drop, others just do NOTHING. That is what happening most of the time. You press your mouse, possibly move it on one pixel and game registeres this as DragNDrop which does nothing, you scream at your monitor about lags or whatever

This PR addes default Click action on dragndrop objects. This should not affect default dragndrop objects, they stay interactable as they were before. This just supposed to reduce moments like on the video or even just remove them from the game. If you click your mouse with weapon in your hand, you want to shoot, not to register dragndrop from one tile to another and just do nothing.

This PR affects every atom in the game and action buttons(abilities)

Im not sure if there is testmerge or whatever needed, but i will test it myself more in future on local server i guess. 
There is limited list of objects were avaiable with dragndrop so testing them wont be hard.

## Testing

compiled, shot, put drask into trash bin

## Changelog
:cl:
fix: If object was drag dropped, but there was no function to process that, you will click object instead of doing nothing. It has timer for work, if drag drop was long enough, no click will happen
tweak: Disabling hover-over item tooltips will disable all tooltips on everything
/:cl:

### CODE REVIEW

File: code/_onclick/drag_drop.dm [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
 			return
 
-	INVOKE_ASYNC(over, PROC_REF(MouseDrop_T), src, usr)
+	INVOKE_ASYNC(over, PROC_REF(MouseDrop_T), src, usr, params)
 
 // recieve a mousedrop
-/atom/proc/MouseDrop_T(atom/dropping, mob/user)
+/atom/proc/MouseDrop_T(atom/dropping, mob/user, params)
+	user.ClickOn(dropping, params)
 	return
```
Comment (Sirryan2002) [inline code/_onclick/drag_drop.dm:21]:
```suggestion

```
hangine return

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -910,25 +910,27 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
...
-		return
-	if(!Adjacent(usr))
-		return
-	if(IsFrozen(src) && !is_admin(usr))
-		to_chat(usr, "<span class='boldannounce'>Interacting with admin-frozen players is not permitted.</span>")
-		return
-	if(isLivingSSD(src) && M.client && M.client.send_ssd_warning(src))
-		return
-	show_inv(usr)
+	if(M == usr)
```
Comment (Sirryan2002) [inline code/modules/mob/mob.dm:915]: reverse the check here and nest `usr.ClickOn(M, params)` under it
Then you can unindent everything below this line

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -910,25 +910,27 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
...
+		if(isliving(M))
+			var/mob/living/L = M
+			if(L.mob_size <= MOB_SIZE_SMALL)
+				return // Stops pAI drones and small mobs (parrots, crabs) from stripping people. --DZD
+		if(!M.can_strip)
+			return
+		if(usr == src)
+			return
+		if(!Adjacent(usr))
+			return
```
Comment (Sirryan2002) [inline code/modules/mob/mob.dm:925]: just condense these into a single if statement

File: code/_onclick/hud/action_button.dm [hunk 4]
```diff
@@ -5,11 +5,13 @@
 	var/ordered = TRUE
 
 /obj/screen/movable/action_button/MouseDrop(over_object)
+	if(over_object == src) // in case something bad happend and game realised we dragged our ability instead of pressing it
+		src.Click()
```
Comment (SteelSlayer) [inline code/_onclick/hud/action_button.dm:9]:
```suggestion
		Click()
```

**Comment by HMBGERDO** [comment]
Alright here is what was updated last
Now we expect TRUE from MouseDrop_T function to make sure we expect drag in that action and to not click object instead of that
I updated all objects with custom MouseDrop_T functions to return TRUE when we expect this.
Delayed parts of functions were replaced with INVOKE_ASYNC, some utility functions were written to work with INVOKE_ASYNC

Tested all objects with custom MouseDrop_T and MouseDrop functions to work as intended, but im just a human, after all, so watch out and keep me informed if something acts weird

List of objects i tested:
  MouseDrop_T:
    /obj/machinery/dna_scannernew
    /obj/machinery/abductor/experiment
    /obj/machinery/bodyscanner
    /obj/machinery/cryopod ?
    /obj/machinery/optable
    /obj/machinery/sleeper
    /obj/machinery/suit_storage_unit ?
    /obj/mecha
    /obj/item
    /obj/structure
    /obj/structure/kitchenspike
    /obj/structure/m_tray
    /obj/structure/c_tray
    /obj/structure/table
    /obj/structure/rack
    /obj/structure/closet
    /obj/machinery/atmospherics/unary/cryo_cell
    /obj/machinery/gibber
    /obj/machinery/smartfridge
    /obj/machinery/photocopier
    /obj/machinery/disposal

  MouseDrop:
    /obj/screen/movable/action_button
    /obj/machinery/iv_drip
    /obj/structure/closet/body_bag
    /obj/item/storage/belt
    /obj/item/storage/pill_bottle
    /obj/item/storage/secure
    /obj/item/storage
    /obj/item/watertank
    /obj/structure/bed/roller
    /obj/structure/chair
    /obj/item/reagent_containers/food/drinks
    /obj/item/reagent_containers/food/drinks/drinkingglass/shotglass
    /obj/item/deck
    /obj/item/shared_storage
    /mob
    /mob/living/silicon/pai
    /mob/living/simple_animal/slime
    /obj/item/desk_bell
    /obj/item/paper_bin
    /obj/item/storage/photo_album
    /obj/item/pda
    /obj/item/slimepotion/fireproof
    /obj/structure/bed/amb_trolley

**Comment by HMBGERDO** [comment]
I think we're ready for testmerge this

**Comment by HMBGERDO** [comment]
so, what is that last commit? This commit prevents cheesing with dragging to attack someone
as expected, if we drag something onto something, but nothing happends, we click the object we dragged
this could be VERY powerful with melee weapon, you drag target from the distance, then go to area next to them and just release the mouse button

last commit adds timer for dragging so if you do your dragging more than 0.2 seconds, no extra clicks will be done
this supposed to prevent players from doing cheesing

this PR was made to fix fast dragndrops when you really expected just click, not to change the entire fight system

**Comment by Peakwastaken** [comment]
this is goated, will improve melee combat a lot

File: code/_onclick/drag_drop.dm [hunk 5]
```diff
@@ -1,19 +1,40 @@
+/atom/MouseDrag()
+	if(!dragging)
+		dragging = TRUE
+		drag_start = world.time
+
+/atom/proc/could_be_lag()
```
Comment (lewcc) [inline code/_onclick/drag_drop.dm:6]: Could you add a comment to this function explaining what its purpose is, and possibly change the function name to be a bit more descriptive about it dealing with click dragging?

File: code/game/alternate_appearance.dm [hunk 6]
```diff
@@ -71,6 +71,8 @@
 	var/list/alternate_appearances //the alternate appearances we own
 	var/list/viewing_alternate_appearances //the alternate appearances we're viewing, stored here to reestablish them after Logout()s
 	//these lists are built as necessary, so atoms aren't all lugging around empty lists
+	var/drag_start = 0
```
Comment (lewcc) [inline code/game/alternate_appearance.dm:74]: Same here, please add some autodoc comments here so it's clear what these do.

Comment (HMBGERDO) [reply code/game/alternate_appearance.dm]: https://github.com/ParadiseSS13/Paradise/pull/21052/commits/d7b1c60523854173f455f567c3d40a4b44cf2902

File: code/_onclick/drag_drop.dm (hunk 5 above)
Comment (HMBGERDO) [reply code/_onclick/drag_drop.dm]: https://github.com/ParadiseSS13/Paradise/pull/21052/commits/d7b1c60523854173f455f567c3d40a4b44cf2902

**Comment by HMBGERDO** [comment]
NEWS!
Inventory objects(pockets, suit storage and other) now supports dragdrop aswell, should fix most I CLICKED MY BATON cases
Movable objects logic now uses `could_be_click_lag` proc instead of checking if we pulled on the same object. Laggy dragdrop logic will be applied only if movable object is locked, otherwise we think you want to move it
Screen tooltips appearing everywhere now checks if player has "Toggle Hover-over Item Tooltips" flag on their preferences. To be honest, people rarely use those tooltips and they close big screen area which became unclickable, which is very uncool. Now you can get rid of them.

**Comment by HMBGERDO** [comment]
Fixed issue when you're trying to climb on the table with item in hand and you hit yourself while climbing.

**Review by S34NW — APPROVED** [review]
Issue free for a bit on live as far as I've seen, lets go.

**Pull request merged**